pippy/ManualPipelineStage.py

```diff
@@ -172,36 +176,55 @@ def get_stage_shapes(
 
 
 class ManualPipelineStage(PipelineStageBase):
+    """
+    A class representing a pipeline stage in a pipeline parallelism setup.
+    This class is created manually by providing a example input (and optionally output)
+    as opposed to the PipelineStage class that is outputed from pipeline().
+    This class extends the `PipelineStageBase` class and can similarly be used
+    in `PipelineScheule`.
+    Args:
+        submodule (nn.Module): The PyTorch module wrapped by this stage.
+        stage_index (int): The ID of this stage.
+        num_stages (int): The total number of stages.
+        device (torch.device): The device where this stage is located.
+        num_microbatches (int): The number of microbatches to use.
+        input_args (Union[torch.Tensor, List[torch.tensor]], optional): The input arguments for the module.
+        output_args (Union[torch.Tensor, List[torch.tensor]], optional): The output arguments for the module.
+        group (dist.ProcessGroup, optional): The process group for distributed training. If None, default group.
```

Comment:
i think its tricky to let group be optional, and have the num_stages vs group_size be ambigious.

(in my case I was passing num_stages =2 and i was forgetting to pass my pp_mesh.group, so ManualStage was complaining that the default group's size wasn't compatible with my num_stages.  Should we just always require group, or is it useful to not require it?

Reply:
we should support devicemesh too.